Automattic/wp-calypso
# Editor: Enable alt text editing for existing images (feature flagged)

Related: #307

This pull request seeks to further progress image editing functionality for existing images in a post. It removes the in-progress editor media global state in favor of a more direct form extraction upon saving changes.

![alt](https://cloud.githubusercontent.com/assets/1779930/15564190/836468b8-22dd-11e6-9fc7-15d2a545a209.png)

_Note that this feature is very much in progress and is only available in development environments._

/cc @robobot3000 I would have liked to have removed `state/ui/editor/media` altogether here. Given my previous remarks on renaming the `imageEditor` directory, I also think it might make sense to bump up to `state/ui/editor/image-editor`.

__Testing instructions:__

Verify that you can edit the alt text for an image by selecting it and choosing the edit button from the inline toolbar.

1. Navigate to the [post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Click Add Media in the editor toolbar
4. Insert an image
5. Click the image to select it
6. In the inline image toolbar, click the Edit button
7. Note that a form is shown, including "Alt text"
8. Enter alt text
9. Click Save
10. Change to the HTML tab to confirm alt text has been updated
